integration-tests/src/test/java/org/apache/fineract/integrationtests/cob/CobPartitioningTest.java

```diff
@@ -0,0 +1,249 @@
+        LOAN_TRANSACTION_HELPER = new LoanTransactionHelper(REQUEST_SPEC, RESPONSE_SPEC);
+
+        TODAYS_DATE = Utils.getLocalDateOfTenant();
+        new BusinessStepHelper().updateSteps("LOAN_CLOSE_OF_BUSINESS", "APPLY_CHARGE_TO_OVERDUE_LOANS", "LOAN_DELINQUENCY_CLASSIFICATION",
+                "CHECK_LOAN_REPAYMENT_DUE", "CHECK_LOAN_REPAYMENT_OVERDUE", "UPDATE_LOAN_ARREARS_AGING", "ADD_PERIODIC_ACCRUAL_ENTRIES",
+                "EXTERNAL_ASSET_OWNER_TRANSFER");
+
+        ASSET_ACCOUNT = accountHelper.createAssetAccount();
+        FEE_PENALTY_ACCOUNT = accountHelper.createAssetAccount();
+        TRANSFER_ACCOUNT = accountHelper.createAssetAccount();
+        EXPENSE_ACCOUNT = accountHelper.createExpenseAccount();
+        INCOME_ACCOUNT = accountHelper.createIncomeAccount();
+        OVERPAYMENT_ACCOUNT = accountHelper.createLiabilityAccount();
+
+        setProperFinancialActivity(TRANSFER_ACCOUNT);
+    }
+
+    private static void setProperFinancialActivity(Account transferAccount) {
+        List<GetFinancialActivityAccountsResponse> financialMappings = FINANCIAL_ACTIVITY_ACCOUNT_HELPER.getAllFinancialActivityAccounts();
+        financialMappings.forEach(mapping -> FINANCIAL_ACTIVITY_ACCOUNT_HELPER.deleteFinancialActivityAccount(mapping.getId()));
+        FINANCIAL_ACTIVITY_ACCOUNT_HELPER.createFinancialActivityAccount(new PostFinancialActivityAccountsRequest()
+                .financialActivityId((long) AccountingConstants.FinancialActivity.ASSET_TRANSFER.getValue())
+                .glAccountId((long) transferAccount.getAccountID()));
+    }
+
+    @Test
+    public void testLoanCOB() throws InterruptedException {
+        try {
+            ExecutorService executorService = Executors.newFixedThreadPool(30);
+            GlobalConfigurationHelper.manageConfigurations(REQUEST_SPEC, RESPONSE_SPEC,
+                    GlobalConfigurationHelper.ENABLE_AUTOGENERATED_EXTERNAL_ID, true);
+            setInitialBusinessDate("2020-03-02");
+
+            int numberOfLoans = 100;
+
+            CountDownLatch countDownLatch = new CountDownLatch(numberOfLoans);
+            for (int i = 0; i < numberOfLoans; i++) {
+                Future<?> unused = executorService.submit(() -> {
+                    Integer clientID = createClient();
+                    Integer loanID = createLoanForClient(clientID);
+                    log.info("**** Client and loan was created: {}", loanID);
+                    if (loanID > 5 && loanID <= 95) {
+                        LOAN_TRANSACTION_HELPER.forecloseLoan("02 March 2020", loanID);
+                    }
+                    countDownLatch.countDown();
+                });
+            }
+            countDownLatch.await();
+            // select loan_status_id, count(*) from m_loan group by loan_status_id;
+
+            log.info("**** Changing business day...");
+            updateBusinessDateAndExecuteCOBJob("2020-03-03");
+            executorService.shutdown();
```

Comment:
I dont really see any assertion whether the COB was executed properly and the loans were processed... :/

Reply:
fixed as we agreed